### PR TITLE
Show level info in Place Page

### PR DESCRIPTION
**Pull request by dvdmrtnz**
This PR adds the floor level info to the Place Page of POIs. This can be specially useful for finding places in big shopping centers.

This is how it will look:

Closes https://github.com/organicmaps/organicmaps/issues/2687

### CODE REVIEW

**Comment by biodranik** [comment]
Thanks! Can you please also check if adding it to the Editor is possible? See data/editor.config and edits for some other fields.

**Comment by dvdmrtnz** [comment]
Okay, so I added the "level" to the editor. However, I don't know how to make it so it is only enabled when the feature is inside a building.

**Comment by biodranik** [comment]
Take a look at `FeatureID Framework::FindBuildingAtPoint(m2::PointD const & mercator) const`

**Comment by strump** [comment]
@biodranik , you wrote

> It would make sense to enable floor editing only if a feature is inside a building.

but sometimes level could be negative if POI is under ground. Even outside any building. 

Sample 1: https://osm.org/node/398315122 - restaurant  on -1 level

Sample 2: https://osm.org/node/9725081763 - hospital on 0 and -1 level. Yes there could be more then one tag value.

**Comment by dvdmrtnz** [comment]
> @biodranik , you wrote
> 
> > It would make sense to enable floor editing only if a feature is inside a building.
> 
> but sometimes level could be negative if POI is under ground. Even outside any building.
> 
> Sample 1: https://osm.org/node/398315122 - restaurant on -1 level
> 
> Sample 2: https://osm.org/node/9725081763 - hospital on 0 and -1 level. Yes there could be more then one tag value.

Yes. Or maybe the feature is drawn slightly outside the building outline as a result of not very precise mapping. Or it could be that the building is not drawn at all because of lack of data. I guess we should maybe allow the "level" property to be always editable? What do you guys think?

**Comment by biodranik** [comment]
Let's discuss editor behavior. 
1. Some countries start floors from zero, some from one. A tourist from UK may use 1 for the 2nd floor in Ukraine. How OSM handles that?
2. There are many one floor buildings where specifying the floor is irrelevant and creates an assumption that the building has several floors. Asking user to fill details here would be a bad UX.

Probably the safest approach is to allow editing floors only in buildings with explicitly set number of floors. And to also check the entered range (but what about the underground minimum?)

And maybe allow to choose an underground floor also for stand alone features without buildings. But what should be visible as a default then?

Regarding the ground floor, maybe we should use different strings in UI for specific locales/languages. Or draw two alternate values together, like:

Ground Floor (1)
1st floor (2)
2nd floor (3)



So looks like the issue is more complex. Let's focus then on displaying first without editing. And then finish editing part separately.

**Comment by dvdmrtnz** [comment]
> So looks like the issue is more complex. Let's focus then on displaying first without editing. And then finish editing part separately.

Okay, then should I drop the last commit, so the rest of the changes can be merged now?

**Comment by biodranik** [comment]
Please create another editor issue + PR based on this PR, with relevant comments. And then remove commit from this branch so this PR can be merged.

**Comment by endim8** [comment]
> Some countries start floors from zero, some from one. A tourist from UK may use 1 for the 2nd floor in Ukraine. How OSM handles that?
The ground floor in OSM should always be 0, regardless of the country (see https://wiki.openstreetmap.org/wiki/Key:level#Ground_floor_number) the editor should just have a note that mentions this, as locale dependent ground floor numbers will get confusing really quickly

File: data/strings/strings.txt [hunk 1]
```diff
@@ -11196,6 +11196,13 @@
     tr = Telefon Ekle
     zh-Hant = 新增電話
 
+  [level]
+    tags = android,ios
+    en = Level
```
Comment (biodranik) [inline data/strings/strings.txt:11201]: be = Паверх

Comment (biodranik) [inline data/strings/strings.txt:11201]: en = Floor
ar = ﺔﻴﺿﺭﺃ
be = Паверх
bg = Етаж
cs = Podlaha
da = Etage
de = Boden
el = Πάτωμα
es = Piso
eu = Solairua
fa = ﻒﮐ
fi = Lattia
fr = Sol
he = המָוֹק
hu = Padló
id = Lantai
it = Pavimento
ja = 床
ko = 바닥
nb = Gulv
nl = Vloer
pl = Piętro
pt = Piso
pt-BR = Piso
ro = Podea
ru = Этаж
sk = Poschodie
sv = Golv
sw = Sakafu
th = พื้น
tr = Zemin
uk = Поверх
vi = Sàn nhà
zh-Hans = 地面
zh-Hant = 地面

**Review by biodranik — CHANGES_REQUESTED** [review]
It would be great to also enable it for Android. @arnaudvergnet can you help please?

**Comment by arnaudvergnet** [comment]
> It would be great to also enable it for Android. @arnaudvergnet can you help please?

I can try to look into this when I find time to work on OM.

**Comment by biodranik** [comment]
@rtsisyk can you port it to Android?

File: data/strings/strings.txt (hunk 1 above)
Comment (matheusgomesms) [reply data/strings/strings.txt:11201]: pt-BR same as pt: Andar

**Comment by dvdmrtnz** [comment]
I just added it to Android too

File: android/src/com/mapswithme/maps/widget/placepage/PlacePageView.java [hunk 2]
```diff
@@ -380,6 +382,9 @@ protected void onFinishInflate()
     mOperator = findViewById(R.id.ll__place_operator);
     mOperator.setOnClickListener(this);
     mTvOperator = findViewById(R.id.tv__place_operator);
+    mLevel = findViewById(R.id.ll__place_level);
+    mLevel.setOnClickListener(this);
```
Comment (strump) [inline android/src/com/mapswithme/maps/widget/placepage/PlacePageView.java:386]: Does label have any onclick listener? As far as I can see only long clicks are handled.

Comment (dvdmrtnz) [reply android/src/com/mapswithme/maps/widget/placepage/PlacePageView.java]: You are right, it's not needed. I just removed it. Thanks!

**Review by vng — APPROVED** [review]

**Pull request merged**